test/SILOptimizer/stdlib/Atomics.swift

```diff
@@ -0,0 +1,76 @@
+// RUN: %target-swift-frontend -O -emit-sil -disable-availability-checking %s | %FileCheck %s
+
+// REQUIRES: synchronization
+
+import Synchronization
+
+//===----------------------------------------------------------------------===//
+// Ensure that we don't destroy the atomic before operations
+//===----------------------------------------------------------------------===//
+
+// CHECK-LABEL: @localLoad
+// CHECK: [[ATOMIC:%.*]] = alloc_stack [lexical] $Atomic<Int>
+// CHECK: [[ATOMIC_PTR:%.*]] = address_to_pointer [[ATOMIC]] : $*Atomic<Int> to $Builtin.RawPointer
+// CHECK-NEXT: [[LOAD:%.*]] = builtin "atomicload_monotonic_Int64"([[ATOMIC_PTR]] : $Builtin.RawPointer) : $Builtin.Int64
+// CHECK-NEXT: [[INT:%.*]] = struct $Int ([[LOAD]] : $Builtin.Int64)
+// CHECK-NEXT: destroy_addr [[ATOMIC]] : $*Atomic<Int>
+// CHECK-NEXT: dealloc_stack [[ATOMIC]] : $*Atomic<Int>
+// CHECK-NEXT: return [[INT]] : $Int
```

Comment:
A couple suggestions on the check lines here:
- add a check-label for the end
- reach for check-next a bit less frequently.  "Is it really key that there not be any other instructions in between?"
- remove duplicative types
- indent the body of the function so that it looks like the textual SIL but with some flavor of check-directives before it

That'd look something like
```
// CHECK-LABEL: sil @localLoad {{.*}} {
// CHECK:         [[ATOMIC:%.*]] = alloc_stack [lexical] $Atomic<Int>
// CHECK:         [[ATOMIC_PTR:%.*]] = address_to_pointer [[ATOMIC]]
// CHECK:         [[LOAD:%.*]] = builtin "atomicload_monotonic_Int64"([[ATOMIC_PTR]] :
// CHECK:         [[INT:%.*]] = struct $Int ([[LOAD]]
// CHECK:         destroy_addr [[ATOMIC]]
// CHECK:         dealloc_stack [[ATOMIC]]
// CHECK:         return [[INT]]
// CHECK-LABEL: } // end sil function 'localLoad'
```

Reply:
I've gone and simplified these a bit let me know what you think

Reply:
Thanks!